src/AbstractFilter.php

```diff
@@ -22,6 +22,7 @@
 
 /**
  * @template TOptions of array
+ * @implements FilterInterface<mixed>
```

Comment:
This is temporary and would normally be another template. Once I've finished with v3 changes, `AbstractFilter` will likely no longer be in use.

Reply:
@gsteel shouldn't this have `@implements FilterInterface<TSomethingNewForSubclassToDeclare>`, or are you saying it would make it worse?

Reply:
For the time being, adding a template there will just explode the psalm issues. Because the plan is to eradicate `AbstractFilter` in favour of `final class SomeFilter implements FilterInterface`